src/action.ts

```diff
@@ -101,12 +101,10 @@ export async function run(
 
 	// Finally send source to builders
 	try {
-		releaseId = await balena.push(
-			inputs.fleet,
-			src,
-			inputs.cache,
-			buildOptions,
-		);
+		releaseId = await balena.push(inputs.fleet, src, inputs.cache, {
+			...buildOptions,
+			noCache: inputs.layerCache,
```

Comment:
This gets set here instead of one of the branches above (master vs PR) because in both workflows the noCache needs to be set as an option. This could be omitted since balena-util will default to false for noCache but then you have to do a conditional if layerCache set layerCache ? Yuck!